pyrokinetics/gk_code/GKInputCGYRO.py

```diff
@@ -623,3 +631,13 @@ def _check_basic_miller(self):
                 break
 
         return is_basic_miller
+
+    def get_ne_te_normalisation(self):
+        # Get electron temp/dens first
+        for i_sp in range(self.data["N_SPECIES"]):
+            if self.data[f"Z_{i_sp+1}"] == -1:
```

Comment:
What about adiabatic electron runs?

Reply:
Good point, will add a flag!

Reply:
@dake0795 in the case where adiabatic electrons are used in GENE, what is the temp/dens they have? I'm guessing density comes from quasi-neutrality but what about the temp?

Reply:
Density is set by quasi-neutrality. I believe the temp is set to be the same as the kinetic species.

Reply:
In the scenario where there are multiple ion species with different temperatures and adiabatic ions (not sure how likely this is) then which temp would be taken?

Reply:
Also this assumes electrons have Z=-1 I guess this may be true for CGYRO?